node/lib/ITranslator.ts

```diff
@@ -0,0 +1,29 @@
+
+/**
+ * Interface that all translators implement.
+ */
+export interface ITranslator {
+    /**
+     * Creates an instance of the device, given device properties obtained from onboarding.
+     * This method is optionally async, meaning it may return either an immediate device
+     * instance or a promise for a device instance.
+     */
+    createDevice(deviceProps: any): IDevice | Promise<IDevice>;
```

Comment:
The bridge has its own API with an `addDevice()` method; the implementation of that method will (after starting the node engine if necessary) load the translator and then call this `createDevice()` method to get the device that is hooked up to the bridge.

---
In reply to: [71910155](https://github.com/opent2t/opent2t/pull/1#discussion_r71910155) [](ancestors = 71910155)